Use of Evidence generating key material for AE identification (#399)
ietf-rats-wg/draft-ietf-rats-corim: draft-ietf-rats-corim.md

- Conducting regular, automated audits and reviews of the system, such as ensuring that users' privileges are correctly configured and that any new code has been audited and approved by independent parties;
- Failing securely in the event of errors to avoid compromising the security of the system.

It is critical that appraisal procedures are auditable and reproducible.
The integrity of code and data during execution is an explicit objective, for example, ensuring that the appraisal functions are executed in an attestable trusted execution environment (TEE).

The integrity of public and private key material and the secrecy of private key material must be ensured at all times.
This includes key material carried in attestation key triples and key material used to verify the authority of triples (such as public keys that identify trusted supply chain actors).
For more detailed information on protecting Trust Anchors, refer to {{Section 12.4 of -rats-arch}}.

> Utilizing the public part of an asymmetric key pair for Evidence generation to identify an Attesting Environment raises privacy considerations that must be carefully considered.

Reviewer: ```suggestion
Utilizing the public part of an asymmetric key pair that is used for Evidence generation to identify an Attesting Environment raises privacy considerations that must be carefully considered.
```

> As it is possible to use the public part of an asymmetric key pair for Evidence generation to be used to identify an Attesting Environment, this method of potentially identifying unique instances of Attesting Environments (and profiling their respective owners) can come with privacy considerations that have to be carefully weighed.

Reviewer 2: The text here is not clear to me?

Are we saying:
1. Asymmetric Key Pair used to sign the Evidence OR
2. Public Part of Asymmetric Key pair is part of Evidence Claim used to identify the Attesting Env?

Reviewer 3: Public keys are PII unless they are for group signing (where the size of the group is privacy preserving).

Author: > Are we saying:
> 
>     Asymmetric Key Pair used to sign the Evidence OR
>     Public Part of Asymmetric Key pair is part of Evidence Claim used to identify the Attesting Env?

Yes. That happens and is a PII issue.

Reviewer 3: Yes, except DAA needs to be updated to support PQC.

> As it is possible to use the public part of an asymmetric key pair for Evidence generation to be used to identify an Attesting Environment, this method of potentially identifying unique instances of Attesting Environments (and profiling their respective owners) can come with privacy considerations that have to be carefully weighed.

Reviewer: That's quite a lengthy sentence :-) I suggest (hopefully non-lossy) compression.
```suggestion
Utilizing the public part of an asymmetric key pair used for Evidence generation to identify an Attesting Environment raises privacy considerations that must be carefully considered.
```

Reviewer: Although, that's not perfect either.  I think what we need to add is the fact that the keypair is unique per attester/device.

The Verifier should use cryptographically protected, mutually authenticated secure channels to all its trusted input sources (Endorsers, RVPs, Verifier Owners).
These links must reach as deep as possible - possibly terminating within the appraisal session context - to avoid man-in-the-middle attacks.

> Minimizing the use of intermediaries is also vital: each intermediary becomes another party that needs to be trusted and therefore factored in the Attesters and Relying Parties' TCBs.

Reviewer 3: ```suggestion
Minimizing the use of intermediaries is also vital: each intermediary becomes another party that might need to be trusted and therefore factored in the Attesters and Relying Parties' TCBs.
```

Reviewer 3: "might" because tunnelling can be used to exclude intermediaries from confidential payloads and blinding can be used to excluded intermediaries from privacy sensitive payloads.
